components/link/index.css

```diff
@@ -10,55 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-@import "../vars/css/components/spectrum-link.css";
+.spectrum-Link {
+  --spectrum-link-animation-duration: var(--spectrum-animation-duration-100);
 
-.spectrum-Link--sizeS {
-  @remapvars {
-    find: /--spectrum-link(.*)-s-/;
-    filter: color;
-    replace: --spectrum-link$1-;
-  }
+  /* Colors */
+  /* Primary */
+  --spectrum-link-text-color-primary-default: var(--spectrum-accent-content-color);
+  --spectrum-link-text-color-primary-hover: var(--spectrum-accent-content-color-hover);
+  --spectrum-link-text-color-primary-active: var(--spectrum-accent-content-color-down);
+  --spectrum-link-text-color-primary-focus: var(--spectrum-accent-content-color-key-focus);
+
+  /* Secondary */
+  --spectrum-link-text-color-secondary-default: var(--spectrum-neutral-content-color-default);
+  --spectrum-link-text-color-secondary-hover: var(--spectrum-neutral-content-color-hover);
+  --spectrum-link-text-color-secondary-active: var(--spectrum-neutral-content-color-down);
+  --spectrum-link-text-color-secondary-focus: var(--spectrum-neutral-content-color-key-focus);
+
+  --spectrum-link-text-color-white: var(--spectrum-white);
+  --spectrum-link-text-color-black: var(--spectrum-black);
 }
 
-.spectrum-Link--sizeM {
-  @remapvars {
-    find: /--spectrum-link(.*)-m-/;
-    filter: color;
-    replace: --spectrum-link$1-;
+/* Windows high contrast mode */
+@media (forced-colors: active) {
+  &--staticBlack {
+    --highcontrast-link-text-color-black: ButtonText;
   }
-}
 
-.spectrum-Link--sizeL {
-  @remapvars {
-    find: /--spectrum-link(.*)-l-/;
-    filter: color;
-    replace: --spectrum-link$1-;
+  &--staticWhite {
+    --highcontrast-link-text-color-white: ButtonText;
   }
-}
 
-.spectrum-Link--sizeXL {
-  @remapvars {
-    find: /--spectrum-link(.*)-xl-/;
-    filter: color;
-    replace: --spectrum-link$1-;
+  .spectrum-Link--secondary {
+    color: LinkText;
+
+    &:hover {
+      color: LinkText;
+    }
+
+    &:active {
+      color: LinkText;
+    }
+
+    &:focus {
+      color: LinkText;
+    }
   }
 }
 
 .spectrum-Link {
-  font-size: var(--spectrum-link-primary-text-size);
   /* Remove the gray background on active links in IE 10. */
   background-color: transparent;
   /* Remove gaps in links underline in iOS 8+ and Safari 8+. */
   -webkit-text-decoration-skip: objects;
   text-decoration: underline;
-  transition: color var(--spectrum-global-animation-duration-100) ease-in-out;
+  transition: color var(--spectrum-link-animation-duration) ease-in-out;
   outline: none;
   cursor: pointer;
 
-  &:focus-ring {
+  color: var(--spectrum-link-text-color-primary-default);
+
+  &:active {
+    color: var(--spectrum-link-text-color-primary-active);
+  }
+
+  &:focus-visible {
+    color: var(--spectrum-link-text-color-primary-focus);
     text-decoration: underline;
     text-decoration-style: double;
   }
+
+  &:hover {
```

Comment:
Just a thought - sorry I didn't think of this before. In the variations after this, your states are in the order of 

1. hover
2. active 
3. focus 

Do you think it would be super extra tidy if the main component were in that order too (hover - active - focus-visible)?

Reply:
Oh good call. The order of actions is important - Lord Vader Hates Fluffy Animals:

1. link
2. visited
3. hover
4. focus
5. active

Reply:
gotta love a subversive acronym!